core/internal/filestream/filestream.go

```diff
@@ -153,6 +155,9 @@ func (fs *fileStream) Start(
 		runID,
 	)
 
+	// when we resume a run, we need to adjust the start time
+	fs.settings.SetStartTime(startTime)
```

Comment:
`filestream` shouldn't be responsible for resume logic, and we should **not** be mutating the settings object. If we don't want to always use the start time on settings, then `filestream` can have a `startTime` field, and `sender` can choose whether to pass `settings.GetStartTime()` or a value from the resume path.

Reply:
the problem is here: https://github.com/wandb/wandb/blob/main/core/internal/filestream/updatestats.go#L23
if we resume a run the startime is wrong as you init the settings to filestream before we get the final value based on the information we get from the backend (during send_run).
this is a hack and happy to change it but we do need to update the start time if we compute these fields in filestream

Reply:
do we need to update it on the run's settings? where else is it used other than in filestream?

Reply:
the flow is the following:
- during init run_record gets a start_time we do a `send_run` request
- if we resume we get back from the backend the total runtime of the run (we overwrite the original start_time every  resume :/) -> we compute the effective new start time
- this is only than used in this system metric events (without this correction the system metric plots look funky)

Reply:
yes, I understand why we need to use a different start time for resumed runs. My question is whether the Settings object actually needs to be updated or whether `filestream` can simply have a `startTime` field as I suggested. It seems that Settings does *not* need to be updated.

Reply:
yeah i think that would work, and you suggesting adding it to the context?

Reply:
Actually, looking at this closer, why does this happen in `filestream.go` at all? `filestream` never reads the start time. We just need it in the `StatsUpdate` struct (`updatestats.go`). There could be a `startTime` field on `Sender` that's initialized to `settings.GetStartTime()` or to the resumed start time as appropriate.

Reply:
how do you pass it to the `updatestats.go`?

Reply:
`StatsUpdate` is instantiated in exactly once place in `sender.go`. You can have a `startTime` field on `StatsUpdate`.

Reply:
ok let me look at the code again if we can do this is the right place to do it

Reply:
wait but we can't guarantee the order of events coming from the system metric and when we get the correct value of the start time, the only point where it is ordered is in the sender send run request. we could move the init logic to that function

Reply:
if the system metric record is ordered behind the start run request, which it should that would work

Reply:
yeah verified in the handler that the messages are ordered, ok putting a fix